Linux/sgx/buildenv.mk

```diff
@@ -65,7 +65,7 @@ $(shell mkdir -p $(PACKAGE_LIB))
 UBUNTU_CONFNAME:=/usr/include/x86_64-linux-gnu/bits/confname.h
 ifneq ("$(wildcard $(UBUNTU_CONFNAME))","")
 	OS_ID=1
-else ifeq ($(origin NIX_PATH),environment)
+else ifeq ($(origin NIX_STORE),environment)
```

Comment:
It needs to align to the SGX SDK code:
https://github.com/intel/linux-sgx/blob/2ee53db4e8fd25437a817612d3bcb94b66a28373/buildenv.mk#L211

Reply:
Created a companion PR: https://github.com/intel/linux-sgx/pull/773

Reply:
Since https://github.com/intel/linux-sgx/pull/773 is approved, it can be also applied here.